include/mbedtls/ecp.h

```diff
@@ -205,6 +205,9 @@ mbedtls_ecp_point;
  * cardinality is denoted by \p N. Our code requires that \p N is an
  * odd prime as mbedtls_ecp_mul() requires an odd number, and
  * mbedtls_ecdsa_sign() requires that it is prime for blinding purposes.
+ * The default implementation only initializes \p A without setting it to the
+ * authentic value for curves with <code>A = -3</code>(SECP256R1, etc),
+ * so pay attention to \p A when using domain parameters.
```

Comment:
This is clearly an improvement over the existing, but as a general rule documentation shouldn't just say "pay attention" but give concrete guidance. Unfortunately here I'm not sure what users are supposed to do: `mbedtls_cmp_int(grp.A, 0) == 0;` is not precise enough (some curves do have A == 0), the only correct test would be `grp.A.p == NULL`, but the field `p` of `mbedtls_mpi` is private.

So, we have a weird situation where `mbedtls_ecp_group::A` is public, but an user can't make sense of it without accessing a private field...

I think documentation alone is not going to be enough here, and we need a new public function. I'm thinking something like `mbedtls_ecp_group_a_is_minus_3()` returns 1 is A is -3 mod P and returns 0 otherwise. Then the instructions here would be that for Short Weierstrass curves you need to call this function first before trying to use A.

@gilles-peskine-arm wdyt?

Reply:
Note: for 2.28 fields are not private, so we could do without the new function, but OTOH I don't think it would be a problem to add such small function even in an LTS.

Reply:
Looking back, making `A` and `B` public was probably a mistake (mine in https://github.com/Mbed-TLS/mbedtls/pull/4595). A lot of code relies on `P` and `N`, even `G` gets used in ECDSA, but in our code base `A` and `B` are only used in `ecp*.c` and in pkparse to check whether a given list of parameters correspond to a known curve. But it's too late to go back on that in 3.x.

Given that it's public — and even if it wasn't public, for that matter — I'd be tempted to take advantage of the fact that `mbedtls_mpi` can represent negative numbers and represent this special value as -3 (which would be the only supported negative value). I think this wouldn't change the code size at the point of use (just replacing `grp->A.p == 0` by `grp->A.s < 0`), but it would add a few bytes of code where the groups are defined. However, using negative bignums feels wrong: we're trying to get rid of them.

So I guess the best thing is to keep the representation as is. And since we've made it a public interface, as you suggest, create a new public function (which can be `static inline`). We should use that function in `ecp.c` anyway because it's better documentation.

Reply:
>In the LTS, I wouldn't change anything (except the documentation).

Actually, is adding a static inline function really a problem in an LTS? The documentation still needs to tell people what to do, and telling them to access `grp->A.p` feels a bit dirty.

Reply:
No, adding a `static inline` function is fine. I'm not sure it's necessary though. But I wouldn't change `ecp.c` to use it: it's a risk and [you did find a problem that I'd missed](https://github.com/Mbed-TLS/mbedtls/pull/6999#discussion_r1286697700)).